tools/harness-automation/autothreadharness/harness_case.py

```diff
@@ -517,9 +519,56 @@ def _test_bed(self):
             and not (settings.DUT_DEVICE and device[0] == settings.DUT_DEVICE[0])
         ]
         logger.info('Available golden devices: %s', json.dumps(devices, indent=2))
+
+        shield_devices = [
+            shield_device
+            for shield_device in settings.SHIELD_GOLDEN_DEVICES
+            if not self.history.is_bad_golden_device(shield_device[0])
+            and not (settings.DUT2_DEVICE and shield_device[0] == settings.DUT2_DEVICE[0])
+        ]
+        logger.info('Available shield golden devices: %s', json.dumps(shield_devices, indent=2))
         golden_devices_required = self.golden_devices_required
 
-        # for test bed with mixed devices
+        dut_device = ()
+        if settings.DUT_DEVICE:
+            dut_device = settings.DUT_DEVICE
+
+        """check if case need to use shield box and its device order in Testbed page
+        Example:
+         In case script leader_9_2_9.py:
+          case_need_shield = True
+          device_order = [('Router_2', False), ('Commissioner', True), ('Router_1', False), ('DUT', True)]
+         On the TestBed page of the Test Harness, the device sort order for Leader_9_2_9
+           should be like:
+             Router_2
+             Commissioner
+             Router_1
+             DUT
+           The ('Commissioner', True) and ('DUT', True) indicate Commissioner device and DUT2 device should
+           be in the RF-box and choose from SHIELD_GOLDEN_DEVICES and DUT2_DEVICE. Otherwise ('DUT', False) means
+           DUT device is not in RF-box and use DUT_DEVICE. The other roles devices with False should be selected
+           from GOLDEN_DEVICES.
+
+         In case script med_6_3_2.py:
+         case_need_shield = True
+         device_order = ''
+         means no device drag order. DUT2_DEVICE should be applied as DUT and the other gold devices
+         are from GOLDEN_DEVICES.
+        """
+        shield_order = []
+        need_shield = False
+        if self.case_need_shield:
+            need_shield = True
+            if not settings.DUT2_DEVICE:
+                logger.info('Must set DUT2_DEVICE')
+                raise FailError('DUT2_DEVICE must be set in settings.py')
+            if isinstance(self.device_order, list) and self.device_order:
+                shield_order = self.device_order
+                logger.info('case %s devices ordered by %s ', self.case, shield_order)
+            else:
+                logger.info('case %s uses %s as DUT', self.case, settings.DUT2_DEVICE)
```

Comment:
there might be no chance for this `else` to execute?

Reply:
if case_need_shield is True and device_order is null, will need DUT2  act as DUT like case MED_6_3_2 and REED_5_6_7 in which DUT should be put into RF-shield-box